src/basic/macro.h

```diff
@@ -376,6 +376,7 @@ static inline int __coverity_check_and_return__(int condition) {
 
 #define STRV_MAKE(...) ((char**) ((const char*[]) { __VA_ARGS__, NULL }))
 #define STRV_MAKE_EMPTY ((char*[1]) { NULL })
+#define STRV_MAKE_CONST(...) ((const char* const*) ((const char*[]) { __VA_ARGS__, NULL }))
```

Comment:
so far we sidestepped the const/strv issue and just made all strvs non-const all the way... I'd stick to that, since C doesn't allow implicit casts between them... i.e. i'd just not use const on strv ever, even if that slightly sucks

Reply:
Actually no, most of the functions in `conf-parser.[ch]` were doing `const char* const*` thing. And I think this is reasonable in that case, because there are many args, most of which are read-only, and some are not. Having this constified makes it easier to see what is going on.

I added wrote the additions here to conform to surrounding style. If we want to drop all the `const char* const*` stuff, we should discuss that separately.

Reply:
we might want to add a helper:

```c
#define STRV_CONSTIFY(char **l) ({
        char **_l = l;
        (const char *const) _l;
})
```

That'd be a typesafe way to constify an strv. And we could have the reverse.

But I am not sure I like the extra glue this adds for very little gain